### PR TITLE
Template BruteForce on BoundingVolume

### PR DESCRIPTION
**Pull request by aprokop**
This is in preparation to using brute force for multi-dimensional data.

Realistically, it's not really `BoundingVolume`, but rather some `Value`. But that can be changed with APIv2.

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -85,19 +85,24 @@ foreach(_test Callbacks Degenerate ManufacturedSolution ComparisonWithBoost)
...
-    "#define ARBORX_TEST_DISABLE_NEAREST_QUERY\n"
-    "#define ARBORX_TEST_DISABLE_CALLBACK_EARLY_EXIT\n"
-    "#include <tstQueryTree${_test}.cpp>\n"
-  )
-  configure_file(
-    "${CMAKE_CURRENT_BINARY_DIR}/tstQueryTree${_test}_BF.cpp.tmp"
-    "${CMAKE_CURRENT_BINARY_DIR}/tstQueryTree${_test}_BF.cpp" COPYONLY
-  )
-  list(APPEND ARBORX_TEST_QUERY_TREE_SOURCES "${CMAKE_CURRENT_BINARY_DIR}/tstQueryTree${_test}_BF.cpp")
+  foreach(_bounding_volume Box)
```
Comment (dalg24) [inline test/CMakeLists.txt:88]: Why did you change this?

Comment (aprokop) [reply test/CMakeLists.txt:88]: True, I did not have to do it in this PR. This is in preparation for the future testing of BruteForce with multi-dimensional geometries.

Comment (dalg24) [reply test/CMakeLists.txt:88]: Yes

Comment (aprokop) [reply test/CMakeLists.txt]: @dalg24 Done.

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -87,7 +87,9 @@ foreach(_test Callbacks Degenerate ManufacturedSolution ComparisonWithBoost)
   list(APPEND ARBORX_TEST_QUERY_TREE_SOURCES "${CMAKE_CURRENT_BINARY_DIR}/tstQueryTree${_test}_BVH.cpp")
   file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/tstQueryTree${_test}_BF.cpp.tmp"
     "#include <ArborX_BruteForce.hpp>\n"
-    "#define ARBORX_TEST_TREE_TYPES Tuple<ArborX::BruteForce>\n"
+    "#include <ArborX_Box.hpp>\n"
+    "template <class MemorySpace> using ArborX__BruteForce = ArborX::BruteForce<MemorySpace, ArborX::Box>;\n"
+    "#define ARBORX_TEST_TREE_TYPES Tuple<ArborX__BruteForce>\n"
```
Comment (dalg24) [inline test/CMakeLists.txt:92]: Not needed either

Comment (dalg24) [reply test/CMakeLists.txt:92]: oh right

**Review by dalg24 — APPROVED** [review]
